tests/tests/tier0/bluechi-monitor-service-removed/test_bluechi_monitor_service_removed.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import threading
+import time
+
+from typing import Dict
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+
+node_name_foo = "node-foo"
+service_simple = "simple.service"
+
+
+def monitor_service(ctrl: BluechiControllerMachine):
+    result, output = ctrl.run_python(os.path.join("python", "monitor.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    node_foo = nodes[node_name_foo]
+    node_foo.copy_systemd_service(
+        service_simple, "systemd", os.path.join("/", "etc", "systemd", "system"))
+    assert node_foo.wait_for_unit_state_to_be(service_simple, "inactive")
+
+    ctrl.bluechictl.start_unit(node_name_foo, service_simple)
+    nodes[node_name_foo].wait_for_unit_state_to_be(service_simple, "active")
+
+    t = threading.Thread(target=monitor_service, args=(ctrl,))
```

Comment:
Is there a specific reason to start the monitoring script in a thread? Just running the script here and adding the short wait and unit stop in the monitor script (after starting the threads) seems a bit simpler.

Reply:
Yes, it is a blocking call. 
You could move the `assert node_foo.systemctl.get_unit_state(service_simple) == "inactive"` in the `monitor.py` on line 107 (after starting the threads - that should work, I think, and you can remove the threading code from the test py. 
It is just a NIT, though.

Reply:
Fixed